src/dp_flow.c

```diff
@@ -531,6 +540,7 @@ void dp_remove_iface_flows(uint16_t port_id, uint32_t ipv4, uint32_t vni)
 			DPS_LOG_ERR("Iterating flow table failed while removing VM flows", DP_LOG_RET(ret));
 			return;
 		}
+		// what if key->l3_type == RTE_ETHER_TYPE_IPV6?
 		if (flow_val->created_port_id == port_id
 			|| (next_key->vni == vni && flow_val->flow_key[DP_FLOW_DIR_ORG].l3_dst.ip4 == ipv4)
 		) {
```

Comment:
Both above can simply be fixed by adding the condition `l3_type == IPV4` since the code is looking for them specifically.

Reply:
Yes